### DISCUSSION TITLE
Combining Auth Mechanisms not working?

### DISCUSSION THREAD

**Discussion by jimbogithub** [quarkusio/quarkus#24518] [Q&A] [answered]
I'm trying to [combine auth mechanisms](https://quarkus.io/guides/security#combining-authentication-mechanisms), in my case a combination of [embedded](https://quarkus.io/guides/security-properties#embedded-realm-configuration) and [quarkus-smallrye-jwt](https://quarkus.io/guides/security-jwt).  I have both working correctly in independent services, however when combining the two in the same service it appears that only the `bearer` mechanism is supported.

Example configuration:

```
# Normal 'bearer' auth for '/api/*' path:
quarkus.http.auth.permission.api.paths=/api/*
quarkus.http.auth.permission.api.policy=authenticated
quarkus.http.auth.permission.api.auth-mechanism=bearer

# A 'special-user' with 'basic' auth for '/special' path:
quarkus.security.users.embedded.enabled=true
quarkus.security.users.embedded.plain-text=true
quarkus.security.users.embedded.users.special-user=special-password
quarkus.security.users.embedded.roles.special-user=special-role
quarkus.http.auth.policy.special-policy.roles-allowed=special-role               
quarkus.http.auth.permission.special.paths=/special
quarkus.http.auth.permission.special.auth-mechanism=basic
quarkus.http.auth.permission.special.policy=special-policy
```

When I attempt to hit the `/special` endpoint it is still requesting a `bearer` auth, e.g:
```
% curl -v http://localhost:8080/special
*   Trying 127.0.0.1:8080...
* Connected to localhost (127.0.0.1) port 8080 (#0)
> GET /special HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.79.1
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 401 Unauthorized
< www-authenticate: Bearer
< content-length: 0
```

Any thoughts on whether I have some config wrong or this might be a bug appreciated.

**Comment by jimbogithub** [comment, marked as answer]
OK, you need to add `quarkus.http.auth.basic=true`.  Suspect it gets switched off by default when you include `quarkus-smallrye-jwt`.

**Comment by sberyozkin** [comment]
@jimbogithub Indeed it is the case, thanks
